cms/taxonomy/management/commands/sync_topics.py

```diff
@@ -0,0 +1,203 @@
+    """
+    existing_parent = existing_topic.get_parent()
+    existing_parent_id = existing_parent.id if existing_parent else None
+
+    return (
+        fetched_topic["title"] == existing_topic.title
+        and fetched_topic.get("description") == existing_topic.description
+        and not existing_topic.removed
+        and existing_parent_id == fetched_topic["parent_id"]
+    )
+
+
+def _update_topic(existing_topic: Topic, fetched_topic: dict[str, str]) -> None:
+    """Sets: title, description, removed = False (since the topic is present in the external data,
+    it cannot be considered removed). Parent/child changes: If the new parent_id is different from the old one,
+    we do a tree move of the topic to the new parent.
+    """
+    logger.info("Updating existing topic: %s with fetched topic: %s", existing_topic.id, fetched_topic)
+    existing_topic.title = fetched_topic.get("title")
+    existing_topic.description = fetched_topic.get("description")
+    existing_topic.removed = False
+    existing_topic.save()
+
+    # If the topic parent has changed then move the node to match
+    existing_parent_id: str | None = getattr(existing_topic.get_parent(), "id", None)
+    if existing_parent_id != fetched_topic["parent_id"]:
+        logger.warning(
+            "Moving topic %s from parent %s to new parent %s. This will also affect the path of %d subtopics.",
+            existing_topic.id,
+            existing_parent_id,
+            fetched_topic["parent_id"],
+            existing_topic.get_children_count(),
+        )
+        existing_topic.move(_get_topic(fetched_topic["parent_id"]), pos="sorted-child")
+
+
+def _create_topic(fetched_topic: dict[str, str]) -> None:
+    """Creates a Topic object with the given id, title, and description, and parent."""
+    logger.info("Saving new topic %s", fetched_topic)
+    new_topic = Topic(
+        id=fetched_topic["id"], title=fetched_topic["title"], description=fetched_topic.get("description")
+    )
+    if fetched_topic["parent_id"]:
+        parent = _get_topic(fetched_topic["parent_id"])
+        new_topic.save_topic(parent_topic=parent)
+    else:
+        new_topic.save_topic()
+
+
+def _check_for_removed_topics(existing_topic_ids: set[str]) -> None:
+    """Figures out which topics exist in the database but were not returned
+    by the external API in this sync cycle.
+    """
+    existing_topics = _get_all_existing_topic_ids()
+    removed_topics = existing_topics.difference(existing_topic_ids)
+    if removed_topics:
+        logger.warning("WARNING: Found %d removed topic(s)", len(removed_topics))
+    for removed_topic_id in removed_topics:
+        logger.warning("Marking topic %s as removed", removed_topic_id)
+        _set_topic_as_removed(removed_topic_id)
+
+
+def _get_all_existing_topic_ids() -> set[str]:
+    all_topics = Topic.objects.all()
+    topic_ids = {topic.id for topic in all_topics}
+    return topic_ids
```

Comment:
Return just the ids from the DB if you don't need all the data.
```suggestion
    return set(Topic.objects.values_list("id", flat=True))
```

Reply:
Addressed in https://github.com/ONSdigital/dis-wagtail/pull/104/commits/89fb9c9c8c63153bba489e9abedc3084ec285310